packages/codemirror/src/Editor.ts

```diff
@@ -15,6 +15,7 @@ export class Editor extends HTMLWidget {
 
     options(): any {
         return {
+            readOnly: this.readOnly(),
```

Comment:
This should not be needed?

Reply:
That allows the option to be set at the point of instantiation.

Reply:
Your code in the update method also gets called on instantiation?

Reply:
The codemirror instantiation takes place in the enter method: `CodeMirror.fromTextArea`

Then the update method sets the option on the already created instance of codemirror. That seems to work well enough with this readOnly option. It just seems like a good idea to allow codemirror to set the option before *their* instantiation.